include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -195,6 +195,8 @@ struct SessionOptions : Base<OrtSessionOptions> {
   SessionOptions& Add(OrtCustomOpDomain* custom_op_domain);
 
   SessionOptions& DisablePerSessionThreads();
+
+  SessionOptions& SetSessionConfiguration(OrtSessionConfigKey config_key, const char* config_value);
 };
```

Comment:
As this is in a SessionOptions class, I don't think you need 'Session' in the method name. AddConfigEntry maybe?